frontend/apps/reader/modules/readertypeset.lua

```diff
@@ -75,10 +75,15 @@ function ReaderTypeset:onReadSettings(config)
     self:onSetPageMargins(self.unscaled_margins)
     self.sync_t_b_page_margins = self.configurable.sync_t_b_page_margins == 1 and true or false
 
-    -- default to disable TXT formatting as it does more harm than good (the setting is not in UI)
-    self.txt_preformatted = config:readSetting("txt_preformatted")
-                         or G_reader_settings:readSetting("txt_preformatted")
-                         or 1
+    if self.ui.document.is_txt then
+        -- default to disable TXT formatting as it does more harm than good
+        self.txt_preformatted = config:readSetting("txt_preformatted")
+                             or G_reader_settings:readSetting("txt_preformatted")
+                             or 1
+    else
+        -- for other formats, it affects crengine cache
```

Comment:
`-- For other formats than txt, we should keep this setting fixed or it could create multiple cache files`